src/zodbbrowser/state.py

```diff
@@ -88,17 +90,21 @@ def __init__(self, obj, tid=None, _history=None):
         self._load()
 
     def _load(self):
-        self.tid = self.history.lastChange(self.requestedTid)
         try:
+            self.tid = self.history.lastChange(self.requestedTid)
             self.pickledState = self.history.loadStatePickle(self.tid)
             loadedState = self.history.loadState(self.tid)
-        except Exception, e:
-            self.loadError = "%s: %s" % (e.__class__.__name__, e)
-            self.state = LoadErrorState(self.loadError, self.requestedTid)
-        else:
-            self.state = getMultiAdapter((self.obj, loadedState,
-                                         self.requestedTid),
-                                         IStateInterpreter)
+            self.state = getMultiAdapter(
+                (self.obj, loadedState, self.requestedTid), IStateInterpreter)
+            return
+        except HistoryMissingError as error:
+            self.loadError = "Missing history: %s" % \
+                             (error)
+        except POSKeyError as error:
+            self.loadError = "Broken object: %s" % (error)
```

Comment:
This doesn't seem right to me: a "broken object" is an existing persistent object that cannot be loaded because it's class is not importable.

Maybe call it a "Missing object"?

Also, `% (error)` is just a misleading way of writing `% error`.